dcos/httpclient_test.go

```diff
@@ -0,0 +1,60 @@
+package dcos
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDefaultTransportBaseNil(t *testing.T) {
+	c := &DefaultTransport{
+		Base: nil,
+	}
+	rt := c.base()
+
+	assert.ObjectsAreEqual(http.DefaultTransport, rt)
+}
+
+func TestDefaultTransportBase(t *testing.T) {
+	c := &DefaultTransport{
+		Base: &DefaultTransport{Base: nil},
+	}
+	rt := c.base()
+
+	assert.IsType(t, &DefaultTransport{}, rt)
+}
+
+func TestDefaultHTTPClientAuth(t *testing.T) {
+	tokenValue := "TestDefaultHTTPClientAuth-token"
+	store := NewConfigStore(nil)
+	store.Set("core.dcos_acs_token", tokenValue)
+	config := NewConfig(store)
+
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Header.Get("Authorization") == "token="+tokenValue {
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte("TestDefaultHTTPClientAuth - Success"))
+		} else {
+			w.WriteHeader(http.StatusUnauthorized)
+			w.Write([]byte("TestDefaultHTTPClientAuth - Forbidden"))
+		}
+	}))
+	defer s.Close()
+
+	c := NewHTTPClient(config)
+
+	resp, err := c.Get(s.URL)
+	assert.NoError(t, err)
+	assert.Equal(t, 200, resp.StatusCode)
+	respBody, _ := ioutil.ReadAll(resp.Body)
+	assert.Contains(t, string(respBody), "Success")
```

Comment:
nit: it should be enough to just assert on the status code and not have any body, as this is test is about the Authorization header being passed.

Reply:
Y indeed think I’ll remove the body assert and add a descriptive message to the code assertion. 

This test is actually taken from my previous branch and I remember back then the test on status code was sometimes confusing once it fails. So I added the body. 

But let me come up with a description to the test that would fix this problem